products/jbrowse-cli/src/commands/text-index.ts

```diff
@@ -121,82 +136,129 @@ export default class TextIndex extends JBrowseCommand {
-              )
-            })
-            done()
-          },
-        })
-
-        const fileDataStream = await (this.isURL(uri)
-          ? this.createRemoteStream(uri)
-          : fs.createReadStream(path.join(outLocation, uri)))
-
-        const gzStream = uri.endsWith('.gz')
-          ? fileDataStream.pipe(createGunzip())
-          : fileDataStream
-
-        const gff3Stream = gzStream
-          .pipe(gff.parseStream({ parseSequences: false }))
-          .pipe(gffTransform)
-
-        // Add gff3Stream to aggregateStream, and DO NOT send 'end' event on
-        // completion, otherwise this would stop all streams from piping to the
-        // aggregate before completion.
-        aggregateStream = gff3Stream.pipe(aggregateStream, { end: false })
-
-        // If a stream ends we have two options:
-        //  1) it is the last stream, so end the aggregate stream.
-        //  2) it is not the last stream, so add a '\n' to separate streams for the indexer.
-
-        // eslint-disable-next-line no-loop-func
-        gff3Stream.once('end', () => {
-          if (--numStreamsFlowing === 0) {
-            aggregateStream.end()
-          } else {
-            aggregateStream.push('\n')
-          }
-        })
+        yield* this.indexGff3(config, attributes, outLocation, quiet)
+      }
+    }
+  }
+
+  async *indexGff3(
+    config: Track,
+    attributes: string[],
+    outLocation: string,
+    quiet: boolean,
+  ) {
+    const {
+      adapter: {
+        gffGzLocation: { uri },
+      },
+      trackId,
+    } = config
+
+    // progress bar code was aided by blog post at
+    // https://webomnizz.com/download-a-file-with-progressbar-using-node-js/
+    const progressBar = new SingleBar(
+      {
+        format: '{bar} ' + trackId + ' {percentage}% | ETA: {eta}s',
+      },
+      Presets.shades_classic,
+    )
+
+    let fileDataStream
+    let totalBytes = 0
+    let receivedBytes = 0
+    if (this.isURL(uri)) {
+      fileDataStream = await this.createRemoteStream(uri)
+      totalBytes = +(fileDataStream.headers['content-length'] || 0)
+    } else {
+      const filename = path.join(outLocation, uri)
+      totalBytes = fs.statSync(filename).size
+      fileDataStream = fs.createReadStream(filename)
+    }
+
+    if (!quiet) {
+      progressBar.start(totalBytes, 0)
+    }
+
+    fileDataStream.on('data', chunk => {
+      receivedBytes += chunk.length
+      progressBar.update(receivedBytes)
+    })
+
+    const gzStream = uri.endsWith('.gz')
+      ? fileDataStream.pipe(createGunzip())
+      : fileDataStream
+
+    const rl = readline.createInterface({
+      input: gzStream,
+    })
+
+    for await (const line of rl) {
+      if (line.startsWith('#')) {
+        continue
+      } else if (line.startsWith('>')) {
+        break
+      }
+
+      const [seq_id, , type, start, end, , , , col9] = line.split('\t')
+      const locStr = `${seq_id}:${start}..${end}`
+
+      if (type !== 'exon' && type !== 'CDS') {
```

Comment:
Let's not have hardcoded feature types that we ignore. Maybe we can pass this in config